src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -513,6 +541,13 @@ internal static T[] DangerousGetUnderlyingArray<T>(this ImmutableArray<T> array)
         internal static ReadOnlySpan<T> AsSpan<T>(this ImmutableArray<T> array)
             => array.DangerousGetUnderlyingArray();
 
+        internal static ImmutableArray<T> DangerousCreateFromUnderlyingArray<T>(ref T[] array)
+        {
+            var proxy = new ImmutableArrayProxy<T> { MutableArray = array };
+            array = null;
```

Comment:
:thought_balloon: I would normally use `Interlocked.Exchange` here, but if you're in a place where that matters for the code then there are bigger problems.